source/type/type.cpp

```diff
@@ -566,13 +566,16 @@ VJ* MetaVehicleJourney::impl_create_vj(const std::string& uri,
 
     // Desactivating the other vjs. The last creation has priority on
     // all the already existing vjs.
+    // We don't modify validity pattern active only once and shift other than 0
     const auto mask = ~canceled_vp.days;
     for_all_vjs([&] (VehicleJourney& vj) {
+        if (!(vj.validity_patterns[0]->days.count() == 1) && (vj.shift != 0)) {
```

Comment:
Please use enum `base_schedule` explicitly.

Reply:
What has to be implemented (for this patch) is just: don't modify a vj that has a valid base_schedule vp.
You are testing too much things here.